hooks/useClickOutside.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (
+  // eslint-disable-next-line no-unused-vars
+  event: MouseEvent | TouchEvent
+) => void;
+
+export const useClickOutside = <T extends HTMLElement>(
+  handler: useClickOutsideType
+) => {
+  const ref = useRef<T>(null);
+  useEffect(() => {
+    const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'click';
+    const listener = (event: MouseEvent | TouchEvent) => {
+      const el = ref?.current;
+      if (!el || el.contains(event.target as Node)) {
```

Comment:
```suggestion
      if (!ref.current?.contains(event.target as Node)) {
```
`ref` will never be `null` (so no need of optional chaining there), where as `ref.current` will be `null` until the node gets mounted

Reply:
The above modification isn't valid because, as the previous condition `el.contains(event.target as Node)` is checking if node exists, your suggestion `!ref.current?.contains(event.target as Node)` checks for if node doesn't exist. 

I would suggest to keep the original `if` condition as it's more simple to understand and read.

Reply:
oops! my bad. Let's stick with the current code 👍🏻